File: Backend/src/main/java/com/feedle/feedleapi/Networking/AddPostRequest.java
package com.feedle.feedleapi.Networking;

import com.feedle.feedleapi.Models.Post;

public class AddPostRequest extends Request{
    private Post post;

    public AddPostRequest(Post post)
    {
        super(RequestType.AddPost);
        this.post = post;
    }

    public Post getPost() {
        return post;
    }
}

File: FeedleDataTier/FeedleDataTier/Network/AddPostRequest.cs
using System.Text.Json.Serialization;
using FeedleDataTier.Models;

namespace FeedleDataTier.Network
{
    public class AddPostRequest : Request
    {
        
      [JsonPropertyName("post")]
      public Post Post { get; set; }

      public AddPostRequest(Post post) : base(RequestType.AddPost)
      {
          this.Post = post;
      }
    }
}